client/consensus/src/collators/lookahead.rs

```diff
@@ -636,7 +637,13 @@ where
                                 let slot = inherent_providers.slot();
                                 let container_chain_slot_duration = (params.get_current_slot_duration)(parent_header.hash());
 
-                                match try_to_buy_core::<_, _, <<OBlock as BlockT>::Header as HeaderT>::Number, _, CIDP, _, _>(params.para_id, aux_data, inherent_providers, &params.keystore, params.orchestrator_client.clone(), params.orchestrator_tx_pool.clone(), parent_header, params.orchestrator_slot_duration, container_chain_slot_duration).await {
+                                let buy_core_result = if params.solochain {
+                                    // TODO: implement parathread support for solochain
+                                    unimplemented!("Cannot buy core for parathread in solochain")
```

Comment:
We should avoid this panic as it is possible for sudo to register parathread which can basically crash the collator with that assignment. 

I would suggest to break the loop or if that also result in error, print a warning and `continue` on next iteration.

Reply:
*fixed*